Lib/profile/sample.py

```diff
@@ -537,51 +549,95 @@ def _validate_collapsed_format_args(args, parser):
         args.outfile = f"collapsed.{args.pid}.txt"
 
 
+def wait_for_process_and_sample(process_pid, sort_value, args):
+    for attempt in range(MAX_STARTUP_ATTEMPTS):
+        try:
+            sample(
+                process_pid,
+                sort=sort_value,
+                sample_interval_usec=args.interval,
+                duration_sec=args.duration,
+                filename=args.outfile,
+                all_threads=args.all_threads,
+                limit=args.limit,
+                show_summary=not args.no_summary,
+                output_format=args.format,
+                realtime_stats=args.realtime_stats,
+            )
+            break
+        except RuntimeError:
+            if attempt < MAX_STARTUP_ATTEMPTS - 1:
+                print("Waiting for process to start...")
+                time.sleep(STARTUP_RETRY_DELAY_SECONDS)
+            else:
+                raise RuntimeError("Process failed to start after maximum retries")
+
+
 def main():
     # Create the main parser
     parser = argparse.ArgumentParser(
         description=(
```

Comment:
It could be easier to put this description in a single module-level multi-line string?